### PR TITLE
Second round of request handler changes

### PR DESCRIPTION
**Pull request by vladsud**
First, there are a bunch of unrelated changes here, simply because things are broken in master, or implemented improperly where it's hard to go by without fixing them. I may spawn separate PRs for these issues

**Key changes:**
1. IFluidHandleContext no longer implements IFluidRouter. It's request() method is renamed to resolveHandle().
This reduces number of responsibilities for both interfaces and makes it clearer on the purpose of interfaces.
Note: This causes DataStoreRuntime to have both request() & resolveHandle().
In next iteration, they will have different behaviors - DDS URIs will fail to resolve when request comes in through request(). I.e. it would be impossible to request DDS from data store unless data store handler implements such access.
This is equivalent to making DDSs private on a data store class.

2. 'path' is removed from both IFluidHandle & IFluidRouter. It has been marked deprecated for many versions.

3. Introducing handleFromLegacyUri() for back compat. Using it in couple places in this change to scope amount of churn - all places where it's used needs revisit. More on that below.

4. Fixing how we refer to task manager ID and how we access task manager & agent scheduler. In many places code confuses data store ID vs. type (even though they are the same thing).
Exposing task manager on IContainerRuntimeBase. More on future changes here below.

5. Vltava: changing registry creation pattern - hiding component type names, forcing everything to go through factories. Same pattern should be expanded to all of our examples, as name we use in registry not always matches type on a factory.
   - Also removing using package name as component type - that has pretty big impact on bundle sizes, and is backward compat hazard - type name can't change, which is not obvious looking at package.json.

6. Last edited is simplified - async load path is deprecated, with assumption that it is being used on a critical path of container loading, with detached container creation.

7. PureDataObject names are busted due to renames for many users (i.e. componentInitializingFirstTime is still used in code in derived classes, when such method was renamed on base class).

Follow up for # 3 & 4 above:
Next step will be to pass data object dependencies to them directly through scoping mechanism on object creation, and start removing access to "globals" in container. The most obvious example why it's bad is PureDataObject.getService() & getMatchMakerContainerService() implementations. Consumers of those are likely not aware that generateContainerServicesRequestHandler() has to be used by container developer. In fact, getMatchMakerContainerService() is not used in our repo (other than UT)!
Better approach is to pass these dependencies through local (to object) scope, and force fluid object implementation to grab and store such dependencies (using handles) within data object itself, for future use. PureDataObjectFactory.instantiateInstance() can validate required dependencies are present. PR #2950 takes a first step on this path.

Basically I want to move to a system where caller passes scope object when creating data objects. And slowly removing access to ContainerRuntime.containerScope (including copy on IDataStoreContext.scope). Similar, remove most abilities to do requests against container for agents inside container.

This is similar paradigm to "regular" coding practices - globals/singletons are bad, they complicate composability of code, leak side-effects and do not allow to "connect" independent pieces of code easily (using adapters). Components should be like classes - clearly expressing their requirements upfront, and consumers satisfying these requirements.

### CODE REVIEW

**Review by curtisman — APPROVED** [review]
:shipit:

**Review by agarwal-navin — APPROVED** [review]
Looks good to me overall.
